rest/src/main/scala/io/udash/rest/openapi/OpenApiMetadata.scala

```diff
@@ -226,14 +227,21 @@ final case class OpenApiBodyOperation[T](
 final case class OpenApiParamInfo[T](
   @reifyName(useRawName = true) name: String,
   @optional @composite whenAbsentInfo: Opt[WhenAbsentInfo[T]],
+  @isAnnotated[optionalParam] optional: Boolean,
   @reifyFlags flags: ParamFlags,
   @infer restSchema: RestSchema[T]
 ) extends TypedMetadata[T] {
   val whenAbsentValue: Opt[JsonValue] = whenAbsentInfo.flatMap(_.fallbackValue)
-  val hasFallbackValue: Boolean = whenAbsentInfo.fold(flags.hasDefaultValue)(_.fallbackValue.isDefined)
+  val isOptional: Boolean = optional || whenAbsentInfo.fold(flags.hasDefaultValue)(_.fallbackValue.isDefined)
+
+  @bincompat private[rest] def hasFallbackValue: Boolean = isOptional
 
   def schema(resolver: SchemaResolver, withDefaultValue: Boolean): RefOr[Schema] =
     resolver.resolve(restSchema) |> (s => if (withDefaultValue) s.withDefaultValue(whenAbsentValue) else s)
+
+  @bincompat private[rest] def this(
```

Comment:
Good call, are we removing these on merge to `master`?

Reply:
In principle. yes. In practice, it may not happen immediately at merge but later in bulk.